test/components/views/messages/DateSeparator-test.tsx

```diff
@@ -115,15 +125,161 @@ describe("DateSeparator", () => {
+
+            // We should not see any room switching going on (`Action.ViewRoom`)
+            expect(dispatcher.dispatch).not.toHaveBeenCalled();
+        });
+
+        it("should not show jump to date error if we already switched to another room", async () => {
+            // Render the component
+            getComponent();
+
+            // Open the jump to date context menu
+            fireEvent.click(screen.getByTestId("jump-to-date-separator-button"));
+
+            // Mimic the outcome of switching rooms while waiting for the jump to date
+            // request to finish. Imagine that we started jumping to "last week", the
+            // network request is taking a while, so we got bored, switched rooms; we
+            // shouldn't jump back to the previous room after the network request
+            // happens to finish later.
+            jest.spyOn(SdkContextClass.instance.roomViewStore, "getRoomId").mockReturnValue("!some-other-room");
+
+            // Try to jump to "last week" but we want an error to occur and ensure that
+            // we don't show an error dialog for it since we already switched away to
+            // another room and don't care about the outcome here anymore.
+            mockClient.timestampToEvent.mockRejectedValue(new Error("Fake error in test"));
+            const jumpToLastWeekButton = await screen.findByTestId("jump-to-date-last-week");
+            fireEvent.click(jumpToLastWeekButton);
+
+            // Wait the necessary time in order to see if any modal will appear
+            await waitEnoughCyclesForModal({
+                useFakeTimers: true,
+            });
+
+            // We should not see any error modal dialog
+            //
+            // We have to use `queryBy` so that it can return `null` for something that does not exist.
+            expect(screen.queryByTestId("jump-to-date-error-content")).not.toBeInTheDocument();
+        });
+
+        it("should show error dialog with submit debug logs option when non-networking error occurs", async () => {
+            // Render the component
+            getComponent();
+
+            // Open the jump to date context menu
+            fireEvent.click(screen.getByTestId("jump-to-date-separator-button"));
+
+            // Try to jump to "last week" but we want a non-network error to occur so it
+            // shows the "Submit debug logs" UI
+            mockClient.timestampToEvent.mockRejectedValue(new Error("Fake error in test"));
+            const jumpToLastWeekButton = await screen.findByTestId("jump-to-date-last-week");
+            fireEvent.click(jumpToLastWeekButton);
+
+            // Expect error to be shown. We have to wait for the UI to transition.
+            await screen.findByTestId("jump-to-date-error-content");
+
+            // Expect an option to submit debug logs to be shown when a non-network error occurs
+            await screen.findByTestId("jump-to-date-error-submit-debug-logs-button");
```

Comment:
I think we could definitely get away with a `getBy` here but thought I would keep them both as `findBy` to be a little more refactor resistant in case of moving/removing other assertions here. If `findBy` reads as a signal that the UI will change, then perhaps we should change it to `getBy`.

There is some value in using `expect` here to make it clear and mentioned in https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#using-get-variants-as-assertions. I've wrapped them in `expect(...).toBeInTheDocument();` 👍